plugins/modules/proxmox_disk.py

```diff
@@ -804,27 +865,21 @@ def main():
 
     elif state == 'moved':
         try:
-            disk_config = disk_conf_str_to_dict(vm_config[disk])
-            disk_storage = disk_config["storage_name"]
-            if proxmox.move_disk(disk, vmid, vm, vm_config):
-                module.exit_json(changed=True, vmid=vmid,
-                                 msg="Disk %s moved from VM %s storage %s" % (disk, vmid, disk_storage))
+            success, message = proxmox.move_disk(disk, vmid, vm, vm_config)
+            if success:
+                module.exit_json(changed=True, vmid=vmid, msg=message)
             else:
-                module.exit_json(changed=False, vmid=vmid, msg="Disk %s already at %s storage" % (disk, disk_storage))
+                module.exit_json(changed=False, vmid=vmid, msg=message)
```

Comment:
1. `success` should be changed to `changed`.
2. Instead of the `if`, simply have `module.exit_json(changed=changed, vmid=vmid, msg=message)`.